vllm/worker/vineyard_llm_cache.py

```diff
@@ -30,32 +31,69 @@
 
 logger = init_logger(__name__)
 
-class CacheServiceMetrics:
-    hit_tokens: int = 0 # Total number of tokens hit. 
-    total_tokens: int = 0 # Total number of tokens requested. 
-    hit_blocks: int = 0 # Total number of blocks hit. 
-    total_blocks: int = 0  # Total number of blocks requested. 
-    counter: int = 0 # Total number of measurements. 
-    time_query: list = [] # Times used query cache from cache service. 
-    time_load: list = [] # Times used load fetched cache to device memory. 
-    time_reshape: list = [] # Times used reshaping tensors for flash attention KV format. 
-    time_unload: list = [] # Times used move computed KV from device memory. 
-    time_update: list = [] # Times used update computed KV to cache service. 
-    normalized_time_query: list = [] # Times used query cache from cache service normalized by number of tokens. 
-    normalized_time_load: list = [] # Times used load fetched cache to device memory normalized by number of tokens. 
-    normalized_time_reshape: list = [] # Times used reshaping tensors for flash attention KV format normalized by number of tokens. 
-    normalized_time_unload: list = [] # Times used move computed KV from device memory normalized by number of tokens. 
-    normalized_time_update: list = [] # Times used update computed KV to cache service normalized by number of tokens. 
-
-    err_query: int = 0 # Number of query errors.
-    err_async_update_task_queue_full: int = 0 # Number of Full exceptions when enqueuing async update tasks
-
-    lock: threading.Lock = threading.Lock()
-    # The following metrics need to be protected by `lock`
-    time_async_update_queue: list = [] # Queuing delays of async update tasks
-    time_async_update_exec: list = [] # Execution times of async update tasks
-    counter_async_update_updated: list = [] # Number of udpated tokens
-    err_update: int = 0 # Number of update errors.
+class CacheServiceMetrics: 
+
+    def __init__(self, max_size=100):
```

Comment:
this design is kind of hacky. Why can not we used vllm metric style?

Reply:
fix in new commit